Feature/aws deploy

lunara-kim opened the pull request:
AWS에 background와 main app을 배포하는 작업에 대한 파일들 추가

- aws에 docker 컨테이너를 올리기 위한 스크립트 추가
- aws background 컨테이너에 데이터 수집 요청을 보낼 수 있게 수정 
- mem0 경로 문제로 main app 실행이 되지 않는 문제 수정
- AWS 배포 가이드 문서 추가
- 관련 이슈 : #70

lunara-kim committed:
aws에 docker 컨테이너를 올리기 위한 스크립트 추가

- 기존 코드 내 예외처리가 필요한 부분 수정
- aws 업로드를 위한 스크립트 추가
- aws 업로드를 위한 설정 추가

lunara-kim committed:
aws background 컨테이너에 데이터 수집 요청을 보낼 수 있게 수정

- db arn 수정
- background에 요청을 보내는 스크립트 추가

lunara-kim committed:
mem0 경로 문제로 main app 실행이 되지 않는 문제 수정

- mem0 경로 접근 권한 추가

lunara-kim committed:
AWS 배포 가이드 문서 추가

- AWS 배포 시 필요한 내용들에 대한 문서 추가

eunhye1kim commented:
@pjython 님, 현재 frontend 로그인 시 
Firebase: Error (auth/unauthorized-domain).

에러가 발생합니다.

http://54.180.143.217:8000/ 경로에 대한 인증 URL 추가를 부탁 드립니다.

lunara-kim committed:
frontend_new를 frontend로 변경

- aws 테스트 환경 구성을 위해 frontend_new를 frontend로 변경

lunara-kim committed:
새 프론트엔드에서 firebase 인증이 되지 않는 문제 수정

- firebase key를 BASE64로 인코딩한 후 aws secret에 추가
- aws secret을 환경변수에 등록한 뒤 이를 firebase key로 사용할 수 있도록
  추가

eunhye1kim commented:
> @pjython 님, 현재 frontend 로그인 시 Firebase: Error (auth/unauthorized-domain).
> 
> 에러가 발생합니다.
> 
> ~http://54.180.143.217:8000/~ 경로에 대한 인증 URL 추가를 부탁 드립니다.

@pjython 님, 기존 ip 기반 url은 서비스를 재배포 할 때마다 ip가 바뀌기 때문에 유효하지 않습니다.

http://my-app-alb-2098918302.ap-northeast-2.elb.amazonaws.com/

이 곳으로 다시 한번 firebase 등록 부탁 드립니다.

번거롭게 해 드려 죄송합니다. (__)

lunara-kim committed:
Copilot 리뷰사항 반영

- REDIS_URL을 AWS Secret에서 받아오게 수정
- background에서 DB 연결정보를 로그에 출력하게 하지 않도록 수정
- arn을 동적으로 받아오는 부분은 사이드이펙트가 크므로 패스

eunhye1kim commented on deploy.sh:
```diff
@@ -0,0 +1,104 @@
+
+# 환경 변수 설정
+AWS_REGION="ap-northeast-2"
+ECR_REGISTRY="692631066742.dkr.ecr.ap-northeast-2.amazonaws.com"
```
배포 정보 수정 시 side effect가 너무 크기 때문에 이 부분은 부득이하게 유지합니다.

eunhye1kim commented on background/celery_app.py:
```diff
@@ -1,141 +1,143 @@
+logger = logging.getLogger(__name__)
+
+# DB 초기화
+logger.info(f"DATABASE_URL: {os.getenv('DATABASE_URL', 'NOT_SET')}")
```
수정했습니다.

eunhye1kim commented on check-resources.sh:
```diff
@@ -0,0 +1,245 @@
+  "market-timing-calendar/langfuse-host"
+  "market-timing-calendar/fastapi-env"
+  "market-timing-calendar/mem0-api-key"
+  "market-timing-calendar/firebase-service-account-key"
```
잠재적으로 사용될 수 있는 값이라 이 부분 유지합니다.

eunhye1kim commented on aws-celery-cloudwatch.sh:
```diff
@@ -0,0 +1,47 @@
+  --task-definition market-timing-background \
+  --region ap-northeast-2 \
+  --launch-type FARGATE \
+  --network-configuration "awsvpcConfiguration={subnets=[subnet-039ee1cea06ad607d,subnet-07754bbf4bae16102,subnet-09daf5213e683e46e,subnet-0e996f670896dfa76],securityGroups=[sg-0ae736da1af062d9d],assignPublicIp=ENABLED}" \
```
배포 정보 수정 시 side effect가 너무 크기 때문에 이 부분은 부득이하게 유지합니다.

eunhye1kim commented:
이후 코드 추가를 위해 self merge 합니다.
@pjython 님 시간 되실 때 코드 확인 부탁 드립니다.
